doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,661 @@
+    MRU window to summon.
+  - When provided: Always summon the window who's name or ID matches the given
+    `window` value. If no such window exists, then create a new window with that
+    name/id.
+
+When provided _with_ `monitor` and `desktop`, `window` behaves in the following
+ways:
+* `desktop`
+  - `"any"`: Go to the desktop the given window is already on.
+  - `"toCurrent"`: If the window is on another virtual desktop, then move it to
+    the currently active one.
+  - `"onCurrent"`: If the window is on another virtual desktop, then move it to
+    the currently active one.
+* `monitor`
+  - `"any"`: Leave the window on the monitor it is already on.
+  - `"toCurrent"`: If the window is on another monitor, then move it to the
+    currently active one.
+  - `"onCurrent"`: If the window is on another monitor, then move it to the
+    currently active one.
+  - `<int>`: If the window is on another monitor, then move it to the specified
+    monitor.
+
+> NOTE: You read that right, `onCurrent` and `toCurrent` both do the same thing
+> when `window` is provided. They both already know which window to select, the
+> context of moving to the "current" monitor is all that those parameters add.
+
+#### Other properties
+
+Some users would like the terminal to just appear when the global hotkey is
+pressed. Others would like the true quake-like experience, where the terminal
+window "slides-in" from the top of the monitor. Furthermore, some users would
+like to configure the speed at which that dropdown happens. To support this
+functionality, the `globalSummon` action will support the following property:
+
+* `"dropdownDuration": float`
+  - When omitted, `0`, or a negative number: No animation is used
+    when summoning the window. The summoned window is focused immediately where
+    it is.
+  - When a positive number is provided, the terminal will use that value as a
+    duration (in seconds) to slide the terminal into position when activated.
+  - The default would be some sensible value. The pane animation is .2s, so
+    `0.2` might be a resonable default here.
+
+We could have alternatively provided a `"dropdownSpeed"` setting, that provided
+a number of pixels per second. In my opinion, that would be harder for users to
+use correctly. I believe that it's easier for users to mentally picture "I'd
+like the dropdown to last 100ms" vs "My monitor is 1504px tall, so I need to set
+this to 15040 to make the window traverse the entire display in .1s"
+
+Some users might want to be able to use the global hotkey to hide the window
+when the window is already visible. This would let the hotkey act as a sort of
+global toggle for the Terminal window. Others might not like that behavior, and
+just want the action to always bring the Terminal into focus, and do nothing if
+the terminal is already focused. To facilitate both these use cases, we'll add
+the following property:
+
+* `"hideWhenVisible": bool`

```

Comment:
`"hideWhenVisible"` sounds like we're hiding the window whenever it's visible (independent of the key binding). But it seems like what you mean is "if the window is visible, the key binding will hide the window".

Could we rename this to something like `"minimizeOnExit"`? (I guess "exit" could be interpreted as closing the window...hmm...)

Reply:
...uh?

This has to do with _pressing the key again_, not anything about exiting. Or minimizing, actually. It does neither of the two things in your proposed name. :smile:

Reply:
> When this hotkey is pressed, and the terminal window is currently active, minimize the window

It minimizes the window when you press the hotkey again (as opposed to not doing anything).

I just wish there was a more concise way to say "switch between restore and minimized".

Reply:
minimizing is a specific thing, and I am not certain that it will be doing that. minimized windows are "iconified" per the shell, not _hidden_. The window does not cease to exist when it is minimized, but it should when hidden.

Reply:
okay so `minimizeWhenVisible` then?

Reply:
hideIfAlreadyVisible? toggleVisibility?

Reply:
toggleVisibility sounds nice